ONSdigital/dp-frontend-dataset-controller
# Feature/update variable table to use static area type title

### What

- Made the variable names with the flag `is_area_type=true` a static title, as per UI designs
- Removed redundant call to `GetVersionDimensions`
- There's been some refactoring within the `handlers` package so there's a few code moves (mainly lift and shift)

Resolves ticket [5700 - Update variables box to align to UI design](https://trello.com/c/N3rNpd16/5700-update-variables-box-to-align-to-ui-design)

### How to review

- Sense check
- Tests pass
- Review image with `data.json`
- _Optionally_ run the stack locally; using the [cantabular-import](https://github.com/ONSdigital/dp-compose/tree/main/cantabular-import) docker container. 
  - Create a flexible dataset
  - Go to a [dataset landing page](http://localhost:8081/datasets/cantabular-flexible-example/editions/2021/versions/1)
  - Verify that the geography variable (or set any variable) `is_area_type=true` in the database and that the variable table is displaying `Area type` for the title
  - **OR** call me 🤙 and I'll show you 

#### Image
<img width="728" alt="variables table" src="https://user-images.githubusercontent.com/19624419/170461709-fff95b54-b559-422c-ab7c-833e6db88297.png">

#### Data
```json
{
    "_id" : ObjectId("6284b7a492476de8acc62ed2"),
    "dimensions" : [ 
        {
            "label" : "Region",
            "links" : {
                "code_list" : {},
                "options" : {},
                "version" : {}
            },
            "href" : "http://api.localhost:23200/v1/code-lists/region",
            "id" : "region",
            "name" : "geography",
            "variable" : "Region",
            "number_of_options" : 10,
            "is_area_type" : true
        }, 
        {
            "label" : "Age",
            "links" : {
                "code_list" : {},
                "options" : {},
                "version" : {}
            },
            "href" : "http://api.localhost:23200/v1/code-lists/age",
            "id" : "age",
            "name" : "age",
            "variable" : "Age",
            "number_of_options" : 8
        }
    ],
    "import_tasks" : {
        "import_observations" : {
            "total_inserted_observations" : NumberLong(0),
            "state" : "created"
        }
    },
    "id" : "719a91d0-55c5-4c7c-8787-5e38f9900c5d",
    "last_updated" : ISODate("2022-05-18T09:12:58.118Z"),
    "e_tag" : "82b557530e1853c5ae80a18b51f09acb9a7b513d",
    "links" : {
        "dataset" : {
            "href" : "http://dp-dataset-api:22000/datasets/Testing-server-v2",
            "id" : "Testing-server-v2"
        },
        "dimensions" : {},
        "edition" : {
            "href" : "http://localhost:22000/datasets/Testing-server-v2/editions/2021",
            "id" : "2021"
        },
        "job" : {
            "href" : "http://localhost:21800/jobs/e3b44872-0f7d-4854-9ad6-0d8b6750f29b",
            "id" : "e3b44872-0f7d-4854-9ad6-0d8b6750f29b"
        },
        "self" : {
            "href" : "http://localhost:22000/instances/719a91d0-55c5-4c7c-8787-5e38f9900c5d"
        },
        "version" : {
            "href" : "http://localhost:22000/datasets/Testing-server-v2/editions/2021/versions/2",
            "id" : "2"
        }
    },
    "state" : "published",
    "unique_timestamp" : Timestamp(1652865178, 1),
    "type" : "cantabular_flexible_table",
    "edition" : "2021",
    "headers" : [ 
        "cantabular_table", 
        "Region", 
        "Age"
    ],
    "is_based_on" : {
        "id" : "Teaching-Dataset",
        "type" : "cantabular_flexible_table"
    },
    "version" : 2,
    "alerts" : [],
    "release_date" : "2022-05-21T00:00:00.000Z",
    "usage_notes" : [],
    "collection_id" : "b-fa8409fe85f79b852423fb60a5c263857a3f5f87dbf9373a470e0e0632abdfff",
    "downloads" : {
        "csv" : {
            "href" : "http://localhost:23600/downloads/datasets/Testing-server-v2/editions/2021/versions/2.csv",
            "public" : "http://minio:9000/public-bucket/datasets/Testing-server-v2-2021-2.csv",
            "size" : "2179"
        },
        "csvw" : {
            "href" : "http://localhost:23600/downloads/datasets/Testing-server-v2/editions/2021/versions/2.csv-metadata.json",
            "public" : "http://minio:9000/public-bucket/datasets/Testing-server-v2-2021-2.csvw",
            "size" : "745"
        },
        "txt" : {
            "href" : "http://localhost:23600/downloads/datasets/Testing-server-v2/editions/2021/versions/2.txt",
            "public" : "http://minio:9000/public-bucket/datasets/Testing-server-v2-2021-2.txt",
            "size" : "1093"
        }
    }
}
```

### Who can review

Frontend go dev
